components/ws-manager/pkg/manager/create.go

```diff
@@ -252,7 +253,18 @@ func (m *Manager) createPVCForWorkspacePod(startContext *startWorkspaceContext)
 				},
 			},
 		},
-	}, nil
+	}
+
+	if startContext.VolumeSnapshot.PvcSnapshotVolumeName != "" {
```

Comment:
This will be done in a separate PR. What will be done is that we would need to check if that volume snapshot is present in the cluster. If not, then create it and use `handle` to link it to correct snapshot in gcp project.